src/main/java/core/basesyntax/ArrayList.java

```diff
@@ -1,49 +1,119 @@
 package core.basesyntax;
 
+import java.util.NoSuchElementException;
+
 public class ArrayList<T> implements List<T> {
+    private static final String ARRAY_OUT_OF_BOUND_EX = "This index is out of the array range...";
+    private static final String NO_SUCH_ELEMENT_EX = "There is no such element...";
+    private static final int DEFAULT_CAPACITY = 10;
+    private int size = 0;
+    private T[] elements;
+
+    public ArrayList() {
+        elements = (T[]) new Object[DEFAULT_CAPACITY];
+    }
+
+    private T[] listToArray(List<T> list) {
+        T[] listArray = (T[]) new Object[list.size()];
+        for (int i = 0; i < list.size(); i++) {
+            listArray[i] = list.get(i);
+        }
+        return listArray;
+    }
+
+    private T[] reSizeArray() {
```

Comment:
please move private methods below public
```suggestion
    private T[] resizeArray() {
```